platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -121,6 +128,19 @@
      */
     private Thread writingThread;
 
+    /** MerkleDb settings */
+    private static final MerkleDbSettings settings = MerkleDbSettingsFactory.get();
+
+    /** Executor for parallel bucket reads/updates in {@link #endWriting()} */
+    private static final ExecutorService flushExecutor = Executors.newFixedThreadPool(
```

Comment:
Rather than using the convenience method, it might be worth creating the ThreadPoolExecutor directly here, and specifying a bounded queue.  Unbounded work queues in ExecutorService objects are one of the more interesting, and frustrating, ways to crash a system under heavy load.  Adding a reasonable RejectedExecutionHandler (such as CallerRunsPolicy) can further help with ensuring the system is stable at the limits.

Reply:
The queue is unlimited here, but in reality there will be no more jobs than the size of the bucket pool.